apm-agent-plugins/apm-vertx/apm-vertx-4-plugin/src/main/java/co/elastic/apm/agent/vertx_4/VertxEventLoopInstrumentation.java

```diff
@@ -0,0 +1,135 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2021 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.vertx_4;
+
+import co.elastic.apm.agent.impl.transaction.AbstractSpan;
+import co.elastic.apm.agent.sdk.advice.AssignTo;
+import co.elastic.apm.agent.vertx.GenericHandlerWrapper;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+public abstract class VertxEventLoopInstrumentation extends VertxWebInstrumentation {
```

Comment:
This misses the case where `io.vertx.core.Context#runOnContext` or `io.vertx.core.Context#executeBlocking)` is called directly, as `VertxImpl` delegates to `Context`.

This can be reproduced in test when changing `routingContext.vertx().executeBlocking` to `routingContext.vertx().getOrCreateContext().executeBlocking`.

Reply:
Resolved through instrumenting the Context directly instead of the VertxImpl